### ISSUE TITLE
D407 False positive on Google style (convention=all)

### ISSUE THREAD

**Issue by thejcannon** [PyCQA/pydocstyle#523]
Howdy folks!

I'm using `pydocstyle` through `flake8-docstrings` and have set `--docstring-convention=all`, which I'm using because our tooling doesn't want to prescribe a particular style (but if you are using a style, you ought to be using it correctly!).

Given the following code:

```python
    def from_text(cls, text, *, parent=None, header_depth=1):
        """Returns the region from the given convention text.

        Args:
            text: (forwarded to the object constructor)
            parent: (forwarded to the object constructor)
            header_depth: specifies which Markdown header depth to identify the section by.

        """
```

I'm receiving `D407 Missing dashed underline after section`, which is surprising as I assumed the script would be able to deduce this docstring is in Google style.

Versions: `flake8`: 3.8.3, `flake8-docstrings`: 1.5.0, `pydocstyle`: 5.1.1
